### PR TITLE
Refactor: torrent cell and small torrent cell

### LINKED ISSUE #5826: MacOS torrent priority chevron no longer visible
### What is the issue?

I just noticed the up or down chevron that denotes torrent priority is no longer visible in the macOS client.

An assumption, given that it was the last major change to the macOS client, but possibly related to #5147. 

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.1.0-dev (22da17f13f)

### PR DESCRIPTION
**Pull request by Coeur**
I factored TorrentCell and SmallTorrentCell to avoid duplicate code.

### CODE REVIEW

**Review by sweetppro — COMMENTED** [review]
It works, although the padding for the priority icon can get a little funky.
Ive uploaded an alternative PR using auto layout which (I think) is more consistent...
#5856

**Comment by GaryElshaw** [comment]
Just wondering if this might have the added bonus of addressing #5717?

**Review by sweetppro — APPROVED** [review]

**Comment by Coeur** [comment]
@GaryElshaw no, unrelated regression, which will need a separate fix

**Comment by ckerr** [comment]
> @ckerr once verified, we'll merge both: first the present one by sweetppro, then mine for the refactoring to avoid duplicate code. I'll test it today.

@Coeur, I've merged @sweetppro's PR and will merge this one as soon as the conflicts are resolved & you give the go-ahead

**Comment by Coeur** [comment]
@ckerr @sweetppro rebase done, so only the refactor is left, which removes ~89 lines in total.

**Review by sweetppro — APPROVED** [review]
looks good, and works well

File: macosx/Base.lproj/MainMenu.xib
Comment (nevack) [inline macosx/Base.lproj/MainMenu.xib:1]: @Coeur Are changes in this file really needed in the end?

**Review by nevack — APPROVED** [review]
LGTM

File: macosx/Base.lproj/MainMenu.xib
Comment (Coeur) [reply macosx/Base.lproj/MainMenu.xib:1]: Minor improvement to potentially save on memory usage since those fields aren't user editable.

**Comment by ckerr** [comment]
TeamCity CI failure is due to server issues, not due to this PR

**Pull request merged**